src/app/components/algorithms/algorithm-view/algorithm-view.component.ts

```diff
@@ -51,37 +55,62 @@ export class AlgorithmViewComponent implements OnInit, OnDestroy {
     private route: ActivatedRoute,
     private utilService: UtilService,
     private config: ApiConfiguration,
-    public guard: ChangePageGuard
+    public guard: ChangePageGuard,
+    private planqkPlatformLoginService: PlanqkPlatformLoginService
   ) {}
 
   ngOnInit(): void {
-    this.routeSub = this.route.params.subscribe(({ algoId }) => {
-      this.algorithmService.getAlgorithm({ algorithmId: algoId }).subscribe(
-        (algo: AlgorithmDto) => {
-          this.algorithm = algo;
-          this.frontendAlgorithm = JSON.parse(
-            JSON.stringify(algo)
-          ) as AlgorithmDto;
-          let subheading = this.algorithm.computationModel
-            .toString()
-            .toLowerCase();
-          subheading = subheading[0].toUpperCase() + subheading.slice(1);
-          this.links[0] = {
-            heading: this.createBreadcrumbHeader(this.algorithm),
-            subHeading: subheading + ' Algorithm',
-          };
-          this.getApplicationAreasForAlgorithm(algoId);
-          this.getProblemTypesForAlgorithm(algoId);
-          this.getTagsForAlgorithm(algoId);
-          this.fetchRevisions();
-        },
-        () => {
+    this.dataSubscription = this.planqkPlatformLoginService
+      .isLoggedIn()
+      .pipe(
+        mergeMap((loggedIn) => {
+          if (loggedIn) {
+            // Try planqk
+            this.configuredApiUrl = this.config.rootUrl;
```

Comment:
`this.config.rootUrl` can potentially (depending on the source observable) be the value from line 70 at this point, which would override the old value and subvert the error recovery in line 74. If you want to be sure, you need to limit the number of times an observable is consumed with the `take(1)` operator before the `mergeMap`.